content/blog/tips-for-faster-ci-builds/index.md

```diff
@@ -0,0 +1,218 @@
+So it's better to switch it off:
+
+```yaml
+name: Build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+env:
+  # Disable incremental compilation for faster from-scratch builds
+  CARGO_INCREMENTAL: 0
+
+jobs:
+  build:
+    runs-on: ... 
+    steps:
+      ...
+```
+
+## Disable Debug Info
+
+While debug info is valuable for debugging, it significantly increases the size of the `./target` directory, which can harm caching efficiency.
+It's easy to switch off:
+
+```yaml
+env:
+  CARGO_PROFILE_TEST_DEBUG: 0
+```
+
+## Use `cargo nextest`
+
+`cargo nextest` enables parallel test execution, which can substantially speed up your CI process.
+While they claim a 3x speedup over `cargo test`, in CI environments I typically observe around 40%
+because the runners don't have as many cores as a developer machine.
+It's still a nice speedup.
+
+```yaml
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: taiki-e/install-action@nextest
+      - uses: Swatinem/rust-cache@v2
+      - name: Compile
+        run: cargo check --locked
+      - name: Test
+        run: cargo nextest 
+```
+
+## `Cargo.toml` Settings
+
+These release profile settings can significantly improve build times and binary size:
+
+```toml
+[profile.release]
+lto = true
+codegen-units = 1
+```
+
+- LTO (Link Time Optimization) performs optimizations across module boundaries, which can reduce binary size and improve runtime performance.
+- Setting `codegen-units = 1` trades parallel compilation for better optimization opportunities. While this might make local builds slower, it often speeds up CI builds by reducing memory pressure on resource-constrained runners.
+
+## Automate Dependency Updates 
+
+Implement [dependabot](https://docs.github.com/en/code-security/getting-started/dependabot-quickstart-guide) to automate dependency updates.
+Instead of manually creating PRs for updates and waiting for CI, dependabot handles this automatically, creating PRs that you can merge when ready.
+
+## Streamline Release Creation 
+
+[`release-plz`](https://release-plz.ieni.dev/) automates release creation when PRs are merged.
+This GitHub action eliminates the manual work of creating releases and is highly recommended for maintaining a smooth workflow.
+
+## Optimize Your Rust Code
+
+If you've implemented all these optimizations and your builds are still slow, it's time to optimize the Rust code itself. I've compiled many tips in my other blog post [here](/blog/tips-for-faster-rust-compile-times/).
+
+## Conclusion
+
+Remember that each project is unique.
+Start with the easier wins like Swatinem's cache action and `--locked` flag, then progressively implement more advanced optimizations as needed. Monitor your CI metrics to ensure the changes are having the desired effect.
```

Comment:
Since this implies a order to these steps I'd move the ARM tip further to the end right after the Cargo.toml changes.

Reply:
Done